js/ui/codingManager.js

```diff
@@ -462,6 +472,18 @@ const CodingManager = new Lang.Class({
         }
     },
 
+    _overviewHiding: function(overview, session) {
+        session.buttonApp.show();
+        if (session.buttonBuilder)
+            session.buttonBuilder.show();
+    },
+
+    _overviewShowing: function(overview, session) {
+        session.buttonApp.hide();
+        if (session.buttonBuilder)
+            session.buttonBuilder.hide();
+    },
```

Comment:
Thanks for the review Sam. I actually looked at the code and while the feeling is that there is some replication, there is not that much in the end. I would keep it that way without refactoring.